### PR TITLE
Add React Router DOM wrapper

### PR DESCRIPTION
**Pull request by neworld**
This is the bare minimum of the wrapper to work in single page applications. I ported it from [kotlinconf-app](https://github.com/JetBrains/kotlinconf-app). Also, strip some stuff and add a method to build react components inside (needed for passing props). Big thanks to @orangy for initial version.

### CODE REVIEW

File: kotlin-react-router-dom/src/main/kotlin/react/router/dom/routingDsl.kt [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    }
+}
+
+fun RBuilder.routeLink(to: String, replace: Boolean = false, handler: RHandler<RProps>?) = child(LinkComponent::class) {
+    attrs {
+        this.to = to
+        this.replace = replace
+    }
+    handler?.invoke(this)
+}
```
Comment (deviant-studio) [inline kotlin-react-router-dom/src/main/kotlin/react/router/dom/routingDsl.kt]: @neworld  please add `redirect` api
```kotlin
if (!loggedIn){
    redirect("/login")
}
```

Comment (neworld) [reply kotlin-react-router-dom/src/main/kotlin/react/router/dom/routingDsl.kt]: Done. Please test it

**Comment by neworld** [comment]
Friendly ping. @Leonya, could you look at this?

**Comment by Leonya** [comment]
Merged and published, thanks for contributing!

**Comment by neworld** [comment]
🎉

**Pull request merged**